src/Components/Components/src/Auth/AuthenticationStateProvider.razor

```diff
@@ -0,0 +1,37 @@
+@namespace Microsoft.AspNetCore.Components
+@implements IDisposable
+
+<CascadingValue T="Task<IAuthenticationState>" Value="@_currentAuthStateTask" ChildContent="@ChildContent" />
+
+@functions {
+    private Task<IAuthenticationState> _currentAuthStateTask;
+
+    [Inject] private IAuthenticationStateProvider AuthStateProvider { get; set; }
```

Comment:
Why does this need to be written out longhand instead of using `@inject`? Is it to control the visibility?

Reply:
Good point - fixed. The visibility remains the same with `@inject`.